modules/objdetect/include/opencv2/objdetect.hpp

```diff
@@ -860,18 +823,100 @@ class CV_EXPORTS_W QRCodeDetector
+     @param straight_qrcode The optional output image containing rectified and binarized QR code
+     */
+    CV_WRAP cv::String decodeCurved(InputArray img, InputArray points, OutputArray straight_qrcode = noArray());
+
+    /** @brief Both detects and decodes QR code on a curved surface
+
+     @param img grayscale or color (BGR) image containing QR code.
+     @param points optional output array of vertices of the found QR code quadrangle. Will be empty if not found.
+     @param straight_qrcode The optional output image containing rectified and binarized QR code
+     */
+    CV_WRAP std::string detectAndDecodeCurved(InputArray img, OutputArray points=noArray(),
+                                              OutputArray straight_qrcode = noArray());
+};
+
+struct CV_EXPORTS_W_SIMPLE QrWithArucoParams {
+    CV_WRAP QrWithArucoParams();
+
+    /** @brief The minimum allowed pixel size of a QR module in the smallest image in the image pyramid, default 4.f */
+    CV_PROP_RW float minModuleSizeInPyramid;
+
+    /** @brief The maximum allowed relative rotation for finder patterns in the same QR code, default pi/12 */
+    CV_PROP_RW float maxRotation;
+
+    /** @brief The maximum allowed relative mismatch in module sizes for finder patterns in the same QR code, default 1.75f */
+    CV_PROP_RW float maxModuleSizeMismatch;
+
+    /** @brief The maximum allowed module relative mismatch for timing pattern module, default 2.f
+     *
+     * If relative mismatch of timing pattern module more this value, penalty points will be added.
+     * If a lot of penalty points are added, QR code will be rejected. */
+    CV_PROP_RW float maxTimingPatternMismatch;
+
+    /** @brief The maximum allowed percentage of penalty points out of total pins in timing pattern, default 0.4f */
+    CV_PROP_RW float maxPenalties;
+
+    /** @brief The maximum allowed relative color mismatch in the timing pattern, default 0.2f*/
+    CV_PROP_RW float maxColorsMismatch;
+
+    /** @brief The algorithm find QR codes with almost minimum timing pattern score and minimum size, default 0.9f
+     *
+     * The QR code with the minimum "timing pattern score" and minimum "size" is selected as the best QR code.
+     * If for the current QR code "timing pattern score" * scaleTimingPatternScore < "previous timing pattern score" and "size" < "previous size", then
+     * current QR code set as the best QR code. */
+    CV_PROP_RW float scaleTimingPatternScore;
+};
+
+class CV_EXPORTS_W QRCodeDetectorAruco : public QRCodeDetectorBase {
+public:
+    CV_WRAP QRCodeDetectorAruco(const QrWithArucoParams& params = QrWithArucoParams());
+    CV_WRAP QrWithArucoParams getDetectorParameters();
+    CV_WRAP void setDetectorParameters(const QrWithArucoParams& params);
+    CV_WRAP aruco::DetectorParameters getArucoParameters();
+    CV_WRAP void setArucoParameters(const aruco::DetectorParameters& params);
```

Comment:
Documentation?

Reply:
added docs to `aruco::DetectorParameters`